js/ui/codeView.js

```diff
@@ -453,12 +459,24 @@ var CodingSession = new Lang.Class({
     },
 
     _setEffectsEnabled: function(actor, enabled) {
-        let desaturate = actor.get_effect('codeview-desaturate');
-        if (desaturate) {
-            desaturate.enabled = enabled;
+        let effect = actor.get_effect('codeview-effect');
+        if (effect) {
+            effect.enabled = enabled;
         } else {
-            desaturate = new Shell.CodeViewEffect({ enabled: enabled });
-            actor.add_effect_with_name('codeview-desaturate', desaturate);
+            let appInfo = this._shellApp.get_app_info();
+            let shaderEffect = appInfo.get_string(_HACK_SHADER_DESKTOP_KEY);
+            let shaderProto;
+
+            // By default, all apps get Shell.CodeViewEffect
+            if (shaderEffect)
+                shaderProto = _HACK_SHADER_MAP[shaderEffect];
+            else
+                shaderProto = Shell.CodeViewEffect;
+
+            if (shaderProto) {
+                effect = new shaderProto({ enabled: enabled });
```

Comment:
coding style nitpick: In modern JS, you can write `new shaderProto({enabled})`

Reply:
Fixed.